cdc: Allow webhook sink to provide client certificates to the remote webhook server

sherman-grewal opened the pull request:
# Summary
Allow the webhook sink to provide client certificates to the remote webhook server.

Resolves #74230

# Testing

1. Install `certstrap` to help with creating the CA and other required certificates
```console
$ brew install certstrap
``` 
2. Create a CA
```console
$ certstrap init --common-name "ExampleCA"
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Created out/ExampleCA.key
Created out/ExampleCA.crt
Created out/ExampleCA.crl
```
3. Create the certificate for the server
```console
$ certstrap request-cert --domain  "localhost"
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Created out/localhost.key
Created out/localhost.csr
```
4. Create the certificate for the client
```console
$ certstrap request-cert --domain  "client"
Enter passphrase (empty for no passphrase):
Enter same passphrase again:
Created out/client.key
Created out/client.csr
```
5. Sign the certificate for the server
```console
$ certstrap sign localhost --CA ExampleCA
Created out/localhost.crt from out/localhost.csr signed by out/ExampleCA.key
```
6. Sign the certificate for the client
```console
$ certstrap sign client --CA ExampleCA 
Created out/client.crt from out/client.csr signed by out/ExampleCA.key
```
7. Pull the server code in this [file](https://github.com/youngkin/gohttps/blob/master/advserver/server.go), and then run the following in your console
```console
$ go build server.go
$ ./server -host "localhost" -srvcert "<path-to-server-cert>" -cacert "<path-to-ca-cert>" \
-srvkey "<path-to-server-key>" -port 3000 -certopt 4
```
8. Follow the steps to creating a webhook sink [here](https://www.cockroachlabs.com/docs/stable/stream-data-out-of-cockroachdb-using-changefeeds.html#create-a-changefeed-connected-to-a-webhook-sink), but use the following SQL command to create the changefeed
```sql
CREATE CHANGEFEED
    FOR TABLE movr.vehicles
    INTO 'webhook-https://localhost:3000?insecure_tls_skip_verify=true&client_cert=<client-cert>&client_key=<client-key>'
    WITH updated;
```
**Note: The client_cert and client_key must all be passed directly as base64-encoded values**

Release notes (enterprise change): Client certificates may now be provided for the webhook changefeed sink.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/74645)
<!-- Reviewable:end -->

miretskiy reviewed (changes requested):
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74645)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @HonoreDB and @sherman-grewal)

___
*[-- commits, line 5 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74645#-Mt4vA9e5ufm_UDJXFOJ):*
I think you need to amend your change to include some  some additional information.
In particular, we should definitely have a release note here.
___
*[pkg/ccl/changefeedccl/sink_webhook.go, line 362 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/74645#-Mt4vnnW95VKPqY2cMhG:-Mt4vnnW95VKPqY2cMhH:bxcwdej) ([raw file](https://github.com/cockroachdb/cockroach/blob/f60862199f78362c03c32301df3e082a7f5644ab/pkg/ccl/changefeedccl/sink_webhook.go#L362)):*
> ```Go
> 	if err := u.decodeBase64(changefeedbase.SinkParamClientKey, &dialConfig.clientKey); err != nil {
> 		return nil, err
> 	}
> ```

Can we expand e.g. `webhookFeedFactory` (in testfeed_test.go) to also support these params?
For example, we could, perhaps add an option to `webhookTestWithOptions` function to also specify if we want "secure" connection, and have webhookFeedFactory start a secure http server for message ingestion?


<!-- Sent from Reviewable.io -->

Commit:
cdc: Allow webhook sink to provide client certificates to the
remote webhook server

Release note (enterprise change): Client certificates may now be provided
for the webhook changefeed sink.

HonoreDB reviewed (approved):
> Reviewed 6 of 6 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/74645)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @HonoreDB and @sherman-grewal)



<!-- Sent from Reviewable.io -->

sherman-grewal commented:
bors r+
